core/Type/Type.php

```diff
@@ -0,0 +1,101 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ *
+ */
+namespace Piwik\Type;
+
+use Piwik\Cache;
+use Piwik\CacheId;
+use Piwik\Plugin;
+use Piwik\Plugin\Manager as PluginManager;
+use Piwik\Property\PropertySettings;
+
+class Type
+{
+    const ID = '';
+    protected $name = 'General_Property';
+    protected $namePlural = 'General_Properties';
+    protected $description = 'Default type';
+    protected $howToSetupUrl = '';
```

Comment:
I see. But having such a thing will probably create some unnecessary code and complexity. It would maybe make sense to me to ignore/not show the `mobileapp` entry if the plugin was removed (which should happen very rarely). (and we could show a notice to the user in that case to explain)

In other words I don't see the point of having a generic "universal" analytics type.